### PR TITLE
Fix: replay the mission cached items only upon reaching resume waypoint

### PR DESCRIPTION
**Pull request by StefanoColli**
### Solved Problem
Upon mission resume (e.g., after a RTL), the gimbal would immediately assume its previous configuration instead of waiting having reached a mission waypoint. 
Similarly the camera would start triggering from take-off instead of from the last reached mission waypoint

### Solution
Store the index of the last mission item reached before the mission interruption, then replay the cached drone commands only after having reached such mission item

### Alternatives
We could also used the existing `_inactivation_index` variable but it has a slightly different meaning and has a different reset behaviour

### Test coverage
- The gimbal issue has been test in SITL
- The camera issue needs to be flight tested

### CODE REVIEW

File: src/modules/navigator/mission_base.cpp [hunk 1]
```diff
@@ -235,6 +235,7 @@ MissionBase::on_activation()
 	checkClimbRequired(_mission.current_seq);
 	set_mission_items();
 
+	_replay_index = _mission.current_seq;
```
Comment (sfuhrer) [inline src/modules/navigator/mission_base.cpp:238]:
```suggestion
	_mission_activation_index = _mission.current_seq;
```
I always try to name variables as descriptive as possible.

File: src/modules/navigator/mission_base.cpp [hunk 2]
```diff
@@ -293,17 +294,22 @@ MissionBase::on_active()
...
-	// replay gimbal and camera commands immediately after resuming mission
-	if (haveCachedGimbalOrCameraItems()) {
-		replayCachedGimbalCameraItems();
-	}
+	// Replay cached mission commands once the last mission waypoint is re-reached after the mission interruption.
+	// Each replay function also clears the cached items afterwards
+	if (_mission.current_seq > _replay_index)
+	{
+		// replay gimbal and camera commands immediately after resuming mission
+		if (haveCachedGimbalOrCameraItems()) {
```
Comment (sfuhrer) [inline src/modules/navigator/mission_base.cpp:302]: I'm not opinionated for one or the other, but when you do the `replayCachedGimbalCameraItems()` you may have the camera not yet correctly configured when the triggering starts. What's the issue with configuring it already when the Mission is resumed, and not only when the vehicle is at the correct position?

File: src/modules/navigator/mission_base.h [hunk 3]
```diff
@@ -328,6 +328,7 @@ class MissionBase : public MissionBlock, public ModuleParams
 	mission_s _mission;					/**< Currently active mission*/
 	float _mission_init_climb_altitude_amsl{NAN}; 		/**< altitude AMSL the vehicle will climb to when mission starts */
 	int _inactivation_index{-1}; // index of mission item at which the mission was paused. Used to resume survey missions at previous waypoint to not lose images.
+	int _replay_index{-1};					/**< Index of the mission item that will bring the drone back to a mission waypoint */
```
Comment (sfuhrer) [inline src/modules/navigator/mission_base.h:331]:
```suggestion
	int _replay_index{-1};					/**< Index of the mission item that will bring the vehicle back to a mission waypoint */
```
Let's stay system agnostic.

**Review by sfuhrer — COMMENTED** [review]
Thanks for your contribution, it works correctly in my simulation. An alternative would be to add a `_work_item_type != WorkItemType::WORK_ITEM_TYPE_CLIMB` after the check for `is_mission_item_reached_or_completed()`, as it's really only the inserted climb mission item that brings the logic out of order. But your proposed solution seems a bit more robust.

File: src/modules/navigator/mission_base.cpp (hunk 2 above)
Comment (StefanoColli) [reply src/modules/navigator/mission_base.cpp:302]: That's a good point, I have split the replay/checks of gimbal commands from the replay/checks if the camera mode. In this way we can replay the camera commands immediately and the gimbal commands later

**Review by sfuhrer — APPROVED** [review]
Looks good to me!

**Pull request merged**